x/ccv/provider/keeper/relay.go

```diff
@@ -385,3 +404,26 @@ func (k Keeper) EndBlockCCR(ctx sdk.Context) {
 		}
 	}
 }
+
+// getMappedInfractionHeight gets the infraction height mapped from val set ID for the given chain ID
+func (k Keeper) getMappedInfractionHeight(ctx sdk.Context,
+	chainID string, valsetUpdateID uint64) (height uint64, found bool) {
+
+	if valsetUpdateID == 0 {
+		return k.GetInitChainHeight(ctx, chainID)
+	} else {
+		return k.GetValsetUpdateBlockHeight(ctx, valsetUpdateID)
+	}
+}
+
+// getChainIdOrPanic returns the chainID from a recv packet,
+// or panics if the packet was sent on a different channel than any of the established CCV channels.
+func (k Keeper) getChainIdOrPanic(ctx sdk.Context, packet channeltypes.Packet) string {
```

Comment:
I'd just get rid of this method by getting the chain id and explicitly panicing at the top of `OnRecvSlashPacket` and `OnRecvVSCMaturedPacket` and then passing it into where it is needed later in those functions.